app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/view/dataset_editing/MetadataEditingFragment.kt

```diff
@@ -63,7 +64,12 @@ class CategoriesEditingFragment : Fragment() {
                     new = true
                 }
                 if (!new) {
+                    /**
+                     * If the dataset is not new, we add the already existing categories
+                     * and the name of the dataset to the layout and set them as non clickable.
+                     */
                     dataset = dBManagement.getDatasetById(datasetId!!)!!
```

Comment:
Maybe the the not-null assertion operator is not necessary here as you already perform the check above

Reply:
I agree, your idea is correct but it is necessary :( I changed the !! to a ?. for the datasetId because it's cleaner I think but for the call to getDSById, I think we can consider that we always receive an existing DSId without problem. Correct me if I am wrong.